lib/semian/sysv_shared_memory.rb

```diff
@@ -0,0 +1,50 @@
+module Semian
+  module SysVSharedMemory #:nodoc:
+    def self.included(base)
+      def base.do_with_sync(*names)
+        names.each do |name|
+          new_name = "#{name}_inner".freeze.to_sym
+          alias_method new_name, name
+          private new_name
+          define_method(name) do |*args, &block|
+            synchronize do
+              method(new_name).call(*args, &block)
+            end
+          end
+        end
+      end
+    end
+
+    def semid
+      -1
+    end
+
+    def shmid
+      -1
+    end
+
+    def synchronize
+      yield if block_given?
```

Comment:
This is not quite what I meant by a no-op synchronize, see my code-example in the other comment. This code should be in the `Simple` classes, not here where it should do the right thing.